azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -275,62 +290,95 @@ private FlowRunner createFlowRunner(final int execId) throws ExecutorManagerExce
     }
 
     // TODO : figure out the metrics
+    // Create the FlowRunner
     final MetricsManager metricsManager = new MetricsManager(new MetricRegistry());
     final CommonMetrics commonMetrics = new CommonMetrics(metricsManager);
     final ExecMetrics execMetrics = new ExecMetrics(metricsManager);
+    final AzkabanEventReporter eventReporter =
+            SERVICE_PROVIDER.getInstance(AzkabanEventReporter.class);
     final FlowRunner flowRunner = new FlowRunner(flow, this.executorLoader,
-        this.projectLoader, this.jobTypeManager, this.azKabanProps, null,
+        this.projectLoader, this.jobTypeManager, this.azKabanProps, eventReporter,
             null, commonMetrics, execMetrics);
     flowRunner.setFlowWatcher(watcher)
         .setJobLogSettings(this.jobLogChunkSize, this.jobLogNumFiles)
         .setValidateProxyUser(this.validateProxyUser)
-        .setNumJobThreads(20);
+        .setNumJobThreads(this.numJobThreadPerFlow);
 
     return flowRunner;
   }
 
-  private void submitFlowRunner(final FlowRunner flowRunner) {
+  /**
+   * Submits the flow to executorService for execution.
+   * @param flowRunner The FlowRunner object.
+   */
+  private void submitFlowRunner(final FlowRunner flowRunner) throws ExecutorManagerException {
     // set running flow, put it in DB
-    this.flowFuture = this.executorService.submit(flowRunner);
+    logger.info("Submitting flow with execution Id " + flowRunner.getExecutionId());
+    final Future<?> flowFuture = this.executorService.submit(flowRunner);
     try {
-      this.flowFuture.get();
-    } catch (final InterruptedException ie) {
-      ie.printStackTrace();
-    } catch (final ExecutionException ee) {
-      ee.printStackTrace();
+      flowFuture.get();
+    } catch (final InterruptedException | ExecutionException e) {
+      logger.error(ExceptionUtils.getStackTrace(e));
+      throw new ExecutorManagerException(e);
     }
   }
 
+  /**
+   * Unzip a file.
+   * @param zipPath The source zip file.
+   * @param dirPath The destination of the zip file content.
+   * @throws IOException
+   */
   private static void unzipFile(final String zipPath,
       final Path dirPath) throws IOException {
     final ZipFile zipFile = new ZipFile(zipPath);
-    System.out.println("source path = " + zipPath);
+    logger.info("source path = " + zipPath);
     final File unzipped = new File(dirPath.toString());
-    System.out.println("unzipped file dir = " + unzipped.toString());
+    logger.info("unzipped file dir = " + unzipped.toString());
     Utils.unzip(zipFile, unzipped);
     zipFile.close();
```

Comment:
zipFile.close should be in finally block. Why don't you use try with resources? https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html 
Throw ExecutorManagerException in case of any exception.